I'm getting a ton of TypeScript errors that look like this now (see CI):

```
Error: bin/plugin/lib/git.js(20,20): error TS2349: This expression is not callable.
  Type 'typeof import("/home/runner/work/gutenberg/gutenberg/node_modules/simple-git/typings/index")' has no call signatures.
```

They happen on lines where I try to instantiate `simple-git` like this:

```js
const SimpleGit = require( 'simple-git' );
const git = SimpleGit(); // <-- error
```

@jsnajdr Any chance this is related to https://github.com/WordPress/gutenberg/pull/28879?

--- a/packages/env/lib/download-sources.js
+++ b/packages/env/lib/download-sources.js
@@ -120,6 +120,7 @@ async function downloadGitSource( source, { onProgress, spinner, debug } ) {
 			'--depth': '1',
 			'--no-single-branch': null,
 		} );
+		await git.cwd( source.clonePath );
 	}
 
 	log( 'Fetching the specified ref.' );

--- a/packages/env/lib/download-sources.js
+++ b/packages/env/lib/download-sources.js
@@ -3,7 +3,7 @@
  * External dependencies
  */
 const util = require( 'util' );
-const NodeGit = require( 'nodegit' );
+const SimpleGit = require( 'simple-git' );
 const fs = require( 'fs' );
 const got = require( 'got' );
 const path = require( 'path' );
@@ -100,69 +100,33 @@ async function downloadSource( source, options ) {
 async function downloadGitSource( source, { onProgress, spinner, debug } ) {
 	const log = debug
 		? ( message ) => {
-				spinner.info( `NodeGit: ${ message }` );
+				spinner.info( `SimpleGit: ${ message }` );
 				spinner.start();
 		  }
 		: () => {};
 	onProgress( 0 );
 
-	const gitFetchOptions = {
-		fetchOpts: {
-			callbacks: {
-				transferProgress( progress ) {
-					// Fetches are finished when all objects are received and indexed,
-					// so received objects plus indexed objects should equal twice
-					// the total number of objects when done.
-					onProgress(
-						( progress.receivedObjects() +
-							progress.indexedObjects() ) /
-							( progress.totalObjects() * 2 )
-					);
-				},
-				certificateCheck: () => 0,
-				credentials: ( url, userName ) => {
-					try {
-						return NodeGit.Cred.sshKeyFromAgent( userName );
-					} catch {
-						return NodeGit.Cred.defaultNew();
-					}
-				},
-			},
-		},
-	};
-
 	log( 'Cloning or getting the repo.' );
-	const repository = await NodeGit.Clone(
-		source.url,
-		source.clonePath,
-		gitFetchOptions
-	).catch( () => {
-		log( 'Repo already exists, get it.' );
-		return NodeGit.Repository.open( source.clonePath );
-	} );
+	const git = SimpleGit();
+
+	const isRepo =
+		fs.existsSync( source.clonePath ) &&
+		( await git.cwd( source.clonePath ).checkIsRepo() );
+
+	if ( isRepo ) {
+		log( 'Repo already exists, using it.' );
+	} else {
+		await git.clone( source.url, source.clonePath, {
+			'--depth': '1',
+			'--no-single-branch': null,
+		} );
+	}
 
 	log( 'Fetching the specified ref.' );
-	const remote = await repository.getRemote( 'origin' );
-	await remote.fetch( source.ref, gitFetchOptions.fetchOpts );
-	await remote.disconnect();
-	try {
-		log( 'Checking out the specified ref.' );
-		await repository.checkoutRef(
-			await repository
-				.getReference( 'FETCH_HEAD' )
-				// Sometimes git doesn't update FETCH_HEAD for things
-				// like tags so we try another method here.
-				.catch(
-					repository.getReference.bind( repository, source.ref )
-				),
-			{
-				checkoutStrategy: NodeGit.Checkout.STRATEGY.FORCE,
-			}
-		);
-	} catch ( error ) {
-		log( 'Ref needs to be set as detached.' );
-		await repository.setHeadDetached( source.ref );
-	}
+
+	await git.fetch( 'origin', source.ref );
+	log( 'Checking out the specified ref.' );
+	await git.checkout( source.ref );
 
 	onProgress( 1 );
 }

--- a/packages/env/package.json
+++ b/packages/env/package.json
@@ -39,9 +39,9 @@
 		"got": "^10.7.0",
 		"inquirer": "^7.1.0",
 		"js-yaml": "^3.13.1",
-		"nodegit": "^0.27.0",
 		"ora": "^4.0.2",
 		"rimraf": "^3.0.2",
+		"simple-git": "^2.34.2",
 		"terminal-link": "^2.0.0",
 		"yargs": "^14.0.0"
 	},

